framework/core/js/src/admin/components/AdminPage.tsx

```diff
@@ -137,9 +141,34 @@ export default abstract class AdminPage<CustomAttrs extends IPageAttrs = IPageAt
       return entry.call(this);
     }
 
-    const { setting, ...attrs } = entry;
+    const { setting, json, refreshAfterSaving, ...attrs } = entry;
 
-    return <FormGroup bidi={this.setting(setting)} {...attrs} />;
+    const originalBidi: (value?: string) => any = this.setting(setting);
+    let bidi: (value?: string) => any;
+
+    if (json) {
```

Comment:
Why do this on the frontend and not the server?

Reply:
because if we do something on the server about setting types, I would prefer a more solid implementation (I really want to go for something like: https://github.com/spatie/laravel-settings)

Don't think this'll be a 2.x thing though.